api/src/main/java/org/apache/iceberg/DataFile.java

```diff
@@ -80,7 +80,8 @@ static StructType getType(StructType partitionType) {
         UPPER_BOUNDS,
         KEY_METADATA,
         SPLIT_OFFSETS,
-        EQUALITY_IDS
+        EQUALITY_IDS,
+        ManifestFile.SPEC_ID.asOptional()
```

Comment:
Current Approach to getting Spec ID into metadata tables

Reply:
Why optional?

Reply:
This is optional so Avro won't fail when we attempt to read a file which does not contain this field, this makes it so we don't have to modify the reader and can still have a slot for the SpecID. You can check some of my older commits for some other ways I tried to do this which involved a bit more ... grossness. 

https://github.com/apache/iceberg/pull/1421/commits/4f16c8cf86dc808fb1e7015982827649d3e377f8

Reply:
Note to myself @RussellSpitzer : remove this

Reply:
Should we document why this is optional @RussellSpitzer or do we prefer to leave the code more pristine and maybe document elsewhere. Definitely something I would imagine forgetting over time.